kooplex-proxy/proxy/crud.py

```diff
@@ -1,33 +1,79 @@
 from sqlalchemy.orm import Session
-from sqlalchemy import text
+from sqlalchemy.ext.declarative import DeclarativeMeta
 
 import models
-from database import engine
+from database import ALLOWED_SCHEMAS
 
 
-def get_country_samples(db: Session, skip: int = 0, limit: int = 100):
-    return db.query(models.MViewCountrySamples).offset(skip).limit(limit).all()
+def schema_changing(model: DeclarativeMeta, endpoint_name: str, endp_schema_key: str):
+    current_schema = model.get_schema()
+    if not endp_schema_key:
+        endp_schema_key = ALLOWED_SCHEMAS.get('schema_1')
+    if endp_schema_key not in ALLOWED_SCHEMAS:
+        values = '/'.join(list(ALLOWED_SCHEMAS.values()))
+        print(f"This schema is unavailable for using, please use one of these schemas: {values}")
+        return 1
+    else:
+        endp_schema = ALLOWED_SCHEMAS[endp_schema_key]
+        if current_schema != endp_schema:
+            model.set_schema(schema_name=endp_schema)
+            print(f"-- Changing schema from '{current_schema}' to '{endp_schema}' for the endpoint: {endpoint_name} --")
+            return 0
+        else:
+            return 0
 
 
-def get_human_meta_mv(db: Session, skip: int = 0, limit: int = 100):
-    return db.query(models.MViewHumanMetaMv).offset(skip).limit(limit).all()
+def get_country_samples(db: Session, skip: int = 0, limit: int = 100, endp_schema_key=''):
```

Comment:
You did the right thing on line 8 "endp_schema_key: str". What is the reason to do it differently on line 26 and all the other functions? (e.g. endp_schema_key='')

Reply:
here
def schema_changing(model: DeclarativeMeta, endpoint_name: str, endp_schema_key: str):
the arguments are necessary for the definition
here
def get_country_samples(db: Session, skip: int = 0, limit: int = 100, endp_schema_key=''):
they are optional
I don't know how is better, we can make it like this
def get_country_samples(db: Session, endp_schema_key: str, skip: int = 0, limit: int = 100)

Reply:
by this way db and endp_schema_key will be required parameters and skill with limit are optional

Reply:
I think that the 2nd one is better.

Reply:
Sorry, I should be clear. I think that "def get_country_samples(db: Session, endp_schema_key: str, skip: int = 0, limit: int = 100)" is better.